godot_project/project_workspace/workspace_context/structure_context/selection_info.gd

```diff
@@ -337,3 +358,74 @@ class SetVirtualObjectPositionHelper:
 		}
 		var snapshot_name: String = MESSAGE_PER_TYPE[_type]
 		_structure_context.workspace_context.snapshot_moment(snapshot_name)
+
+
+class SetVirtualObjectRotationHelper:
+	enum Type {
+		SHAPE,
+		MOTOR,
+	}
+	
+	var changed_signal: Signal
+	
+	var _type: Type
+	var _structure_context: StructureContext
+	
+	func _init(out_structure_context: StructureContext) -> void:
+		if out_structure_context.nano_structure is NanoShape:
+			_type = Type.SHAPE
+		elif out_structure_context.nano_structure is NanoVirtualMotor:
+			_type = Type.MOTOR
+		else:
+			assert(false, "Unknown type of Virtual Object")
+		changed_signal = out_structure_context.nano_structure.transform_changed
+		_structure_context = out_structure_context
+	
+	
+	func get_rotation_euler() -> Vector3:
+		var basis: Basis = _structure_context.nano_structure.get_transform().basis
+		var euler: Vector3 = basis.get_euler()
+		if _type == Type.MOTOR:
+			# Virtual Motors faces RIGHT instead of FORWARD
+			# because of this we need a special conversion:
```

Comment:
What implications would it have to make the virtual motors face forward like everything else instead?

Reply:
Might be wrong but I think it's matter of rotating model in blender and exporting it again, seems like a bit more elegant solution

Reply:
if we change the model we have to update the OpenMM simulation to know the new axis, no big deal. But my concern is loading an already created file will put the motors pointing to the wrong direction